packages/commerce-sdk-react/src/hooks/utils.ts

```diff
@@ -42,22 +45,24 @@ export const updateCache = <Action extends ShopperCustomersMutationType>(
                 : item === cacheQuery.queryKey[index]
         )
 
-    // STEP 1. Update data inside query cache for the matching queryKeys
-    queryKeysMatrix[action](params, response)?.update?.map((queryKey: any) => {
-        queryClient.setQueryData(queryKey, response)
-    })
+    if (queryKeysMatrix && queryKeysMatrix[action]) {
+        // STEP 1. Update data inside query cache for the matching queryKeys
+        queryKeysMatrix[action]?.(params, response)?.update?.map((queryKey: any) => {
+            queryClient.setQueryData(queryKey, response)
+        })
 
-    // STEP 2. Invalidate cache entries with the matching queryKeys
-    queryKeysMatrix[action](params, response)?.invalidate?.map((queryKey: any) => {
-        queryClient.invalidateQueries({
-            predicate: (cacheQuery: any) => isMatchingKey(cacheQuery, queryKey)
+        // STEP 2. Invalidate cache entries with the matching queryKeys
+        queryKeysMatrix[action]?.(params, response)?.invalidate?.map((queryKey: any) => {
+            queryClient.invalidateQueries({
+                predicate: (cacheQuery: any) => isMatchingKey(cacheQuery, queryKey)
+            })
         })
-    })
 
-    // STEP 3. Remove cache entries with the matching queryKeys
-    queryKeysMatrix[action](params, response)?.remove?.map((queryKey: any) => {
-        queryClient.removeQueries({
-            predicate: (cacheQuery: any) => isMatchingKey(cacheQuery, queryKey)
+        // STEP 3. Remove cache entries with the matching queryKeys
+        queryKeysMatrix[action]?.(params, response)?.remove?.map((queryKey: any) => {
+            queryClient.removeQueries({
+                predicate: (cacheQuery: any) => isMatchingKey(cacheQuery, queryKey)
+            })
         })
-    })
+    }
```

Comment:
The guard around these update actions that checks for existence of `queryKeysMatrix?.[action]` ... should this terminate in an `} else { throw new Error(`message`)}` to warn the client implementation of invalidity?

Reply:
Or we can keep the if-else and throw an error...

Reply:
Yup, I think I'm saying the same thing :)

Reply:
Users can implement the mutation hooks with a string which could lead to an error or otherwise use `ShopperOrdersMutationType` or `ShopperCustomersMutationType` which most likely will protect from using a nonexistent mutation key